doozer/doozerlib/cli/get_nightlies.py

```diff
@@ -187,12 +194,31 @@ async def find_rc_nightlies(runtime: Runtime, arches: Set[str], allow_pending: b
 
     async def _find_nightlies(_arch: str):
         # retrieve the list of nightlies from the release-controller
-        rc_url: str = f"{rc_api_url(tag_base, _arch)}/tags"
+        rc_url: str = f"{rc_api_url(tag_base, _arch, private_nightly)}/tags"
         logger.info(f"Reading nightlies from {rc_url}")
 
-        async with aiohttp.ClientSession() as session:
-            async with session.get(rc_url) as resp:
-                data = await resp.json()
+        # Get the token
+        oc_token_result = subprocess.run(["oc", "whoami", "-t"], capture_output=True, text=True)
+
+        if private_nightly:
+            # Extract the token from the command output
+            if oc_token_result.returncode == 0:  # Check if the command was successful
+                token = oc_token_result.stdout.strip()
+                if not token:
+                    raise ValueError("Token empty, might not be logged in to correct cluster")
+            else:
+                raise ValueError("Error while trying to get the token for reading private nightlies")
+
+            async with aiohttp.ClientSession() as session:
+                headers = {
+                    "Authorization": f"Bearer {token}"
+                }
+                async with session.get(rc_url, headers=headers) as resp:
+                    data = await resp.json()
```

Comment:
since this part is being repeated, we can populate `headers` var in `if private_nightly` and send headers in the common code block.

Reply:
done